### ISSUE TITLE
libbz2.so.1.0 cannot open shared object file

### ISSUE THREAD

**Issue by Reedwarbler** [google/nucleus#11]
I met an issue after finished installing nucleus using command (no error happens):
`pip install --user google-nucleus`

However, when I try to run the following demo code, I got error:
```
from nucleus.io import vcf
sf_vcf="test_nist.b37_chr20_100kbp_at_10mb.vcf.gz"
with vcf.VcfReader(sf_vcf) as reader:
    print('Sample names in VCF: ', ' '.join(reader.header.sample_names))
```

`libbz2.so.1.0: cannot open shared object file: No such file or directory`

Any suggestion for solving the issue? Thank you!

**Comment by ThomasColthurst** [comment]
One easy suggestion is to try running the demo code from a different directory.  This error can happen if you run the demo from the Nucleus base directory, because then Python tries to load Nucleus from that location rather than from the installed location, and only the installed location has the compiled .so files.

If that doesn't work, you will want to run
$ python
>>> import sys
>>> sys.path
to see where Python is importing Nucleus from.  (Since you used --user, the directory name will probably be based in your home directory and will include "site-packages" at the end.)

Once you have that directory, run
$ ls [import dir]/nucleus/io/python

That directory should contain a bunch of .so files like bedgraph_reader.so, gff_writer.so, etc.  If it doesn't, there was an installation error.

**Comment by Reedwarbler** [comment]
```

**Comment by Reedwarbler** [comment]
I run outside the installation folder, but still got the error.

I saw the .so files:
```
bedgraph_reader.so  bed_writer.so    gff_reader.so            __init__.py                reference.so   unindexed_fasta_reader.so
bedgraph_writer.so  fastq_reader.so  gff_writer.so            __init__.pyc               sam_reader.so  vcf_reader.so
bed_reader.so       fastq_writer.so  indexed_fasta_reader.so  in_memory_fasta_reader.so  sam_writer.so  vcf_writer.so
```

**Comment by pgrosu** [comment]
Just update your `LD_LIBRARY_PATH` and `PYTHONPATH` environmental variables, and try again.

**Comment by Reedwarbler** [comment]
Sorry @pgrosu, how do I update?

**Comment by pgrosu** [comment]
Here are a few links describing it for `LD_LIBRARY_PATH`:

https://www.tecmint.com/understanding-shared-libraries-in-linux/
https://docs.oracle.com/cd/E19455-01/816-0559/chapter2-48927/index.html

The same for `PYTHONPATH`, which is described here:

https://docs.python.org/2/using/cmdline.html#envvar-PYTHONPATH

Make sure you echo them out for (i.e. `echo $LD_LIBRARY_PATH`) to see what they already contain, which you want to keep.  You can test changes like this:

`LD_LIBRARY_PATH=your_DIR:$LD_LIBRARY_PATH PYTHONPATH=... python Your_Program ...`

Basically standard Linux config :)

~p

**Comment by pgrosu** [comment]
@Reedwarbler Which linux distribution do you have (i.e. Ubuntu, Redhat, etc)?  I can point you to where you can get the package for it from.  Alternatively, you can compile it from source, which you can get it from here:

https://sourceforge.net/projects/bzip2/files/

**Comment by xyq013563** [comment]
> Here are a few links describing it for `LD_LIBRARY_PATH`:
> 
> https://www.tecmint.com/understanding-shared-libraries-in-linux/
> https://docs.oracle.com/cd/E19455-01/816-0559/chapter2-48927/index.html
> 
> The same for `PYTHONPATH`, which is described here:
> 
> https://docs.python.org/2/using/cmdline.html#envvar-PYTHONPATH
> 
> Make sure you echo them out for (i.e. `echo $LD_LIBRARY_PATH`) to see what they already contain, which you want to keep. You can test changes like this:
> 
> `LD_LIBRARY_PATH=your_DIR:$LD_LIBRARY_PATH PYTHONPATH=... python Your_Program ...`
> 
> Basically standard Linux config :)
> 
> ~p

Thank you very much! Update the  LD_LIBRARY_PATH definitely fixed all of my problems.
